### PR TITLE
Chart upgrade 2021 02

### PR DESCRIPTION
**Pull request by ubergesundheit**
This PR syncs the helm chart of the upstream prometheus-operator-app using the [git subtree](https://hackmd.io/V-7Z_uaNTUmYWaa5Gsz02A?view) method.

- I've deleted the `hacks` & `patch` directory
- CI switches to `app-build-suite` for metadata generation and linting

### CODE REVIEW

File: helm/prometheus-operator-app/Chart.yaml [hunk 1]
```diff
@@ -34,4 +33,17 @@ annotations:
       url: https://github.com/prometheus-community/helm-charts
     - name: Upstream Project
       url: https://github.com/prometheus-operator/kube-prometheus
-  application.giantswarm.io/values-schema: https://raw.githubusercontent.com/giantswarm/prometheus-operator-app/v[[ .Version ]]/helm/prometheus-operator-app/values.schema.json
```
Comment (QuentinBisson) [inline helm/prometheus-operator-app/Chart.yaml:37]: I think you should add this back

File: helm/prometheus-operator-app/templates/prometheus-operator/deployment.yaml [hunk 2]
```diff
@@ -62,28 +62,28 @@ spec:
             {{- end }}
             - --namespaces={{ $ns | join "," }}
             {{- end }}
-            - --logtostderr=true
             - --localhost=127.0.0.1
+            {{- if .Values.prometheusOperator.prometheusDefaultBaseImage }}
             {{- if (.Values.Installation) }}
-            - --prometheus-config-reloader={{ .Values.Installation.V1.Registry.Domain }}/{{ .Values.prometheusOperator.prometheusConfigReloaderImage.repository }}:{{ .Values.prometheusOperator.prometheusConfigReloaderImage.tag }}
+            - --prometheus-default-base-image={{ .Values.Installation.V1.Registry.Domain }}/{{ .Values.prometheusOperator.prometheusDefaultBaseImage }}
```
Comment (QuentinBisson) [inline helm/prometheus-operator-app/templates/prometheus-operator/deployment.yaml:68]: Once config controller is used for prometheus operator, we will be able to remove this part and just define the prometheusDefaultBaseImage property in the helm chart, same for the alertmanager

File: helm/prometheus-operator-app/Chart.yaml (hunk 1 above)
Comment (ubergesundheit) [reply helm/prometheus-operator-app/Chart.yaml:37]: Check https://github.com/giantswarm/giantswarm-test-catalog/commit/c4778c8a00afeb4b41462de04f57ac72a048dbfe#diff-08c65c1b5ef8384ef0200558777c52f4110afcbbacc880ac5375d555ca97ee3eR27005

app-build-suite adds some of these annotations automatically

File: helm/prometheus-operator-app/templates/prometheus-operator/deployment.yaml (hunk 2 above)
Comment (ubergesundheit) [reply helm/prometheus-operator-app/templates/prometheus-operator/deployment.yaml:68]: Ah thanks for this insight!

Comment (ubergesundheit) [reply helm/prometheus-operator-app/templates/prometheus-operator/deployment.yaml:68]: Can you check a few lines below ? https://github.com/giantswarm/prometheus-operator-app/pull/79/files/dbb6746610a992def63e1234ddf1575ff7dab3dd#diff-e27896bc0794dd5a2ff25c437b60bb597c4b52f02e224d83e61d1bacb7d1f6f3R81

**Review by piontec — APPROVED** [review]
I skipped everything that looked like an upstream change. The rest (CI config and such) looks OK.

**Review by QuentinBisson — APPROVED** [review]
LGTM if upgrades are tested :D

**Comment by ubergesundheit** [comment]
Upgrade works :)

**Comment by QuentinBisson** [comment]
Can you release it?

**Comment by ubergesundheit** [comment]
Sure :) Check https://github.com/giantswarm/prometheus-operator-app/pull/82

**Comment by TheoBrigitte** [comment]
TIL that we must not squash commit when using `git subtree` otherwise we loose history and cannot use `git subtree merge`.

Here is the error I'm facing when I try to upgrade the subtree now
```
$ git subtree merge -m "Upgrade v0.51.2" --squash -P helm/prometheus-operator-app/ upstream
Can't squash-merge: 'helm/prometheus-operator-app' was never added.
```

**Pull request merged**